src/GraphQL.Tests/Types/AllScalarGraphTypeTests.cs

```diff
@@ -0,0 +1,807 @@
+        [InlineData(typeof(BigIntGraphType), 0)]
+        [InlineData(typeof(BigIntGraphType), 1E+25)]
+        public void toAST_ok(Type graphType, object value)
+        {
+            if (graphType == typeof(BigIntGraphType))
+                value = new BigInteger(Convert.ToDecimal(value));
+
+            var g = (ScalarGraphType)graphType.GetConstructor(Type.EmptyTypes).Invoke(null);
+            var types = new Type[]
+            {
+                typeof(byte),
+                typeof(sbyte),
+                typeof(short),
+                typeof(ushort),
+                typeof(int),
+                typeof(uint),
+                typeof(long),
+                typeof(ulong),
+                typeof(BigInteger)
+            };
+
+            foreach (var type in types)
+            {
+                object converted;
+                try
+                {
+                    if (type == typeof(BigInteger))
+                    {
+                        converted = new BigInteger((decimal)Convert.ChangeType(value, typeof(decimal)));
+                    }
+                    else
+                    {
+                        converted = Convert.ChangeType(value, type);
+                    }
+                }
+                catch
+                {
+                    continue;
+                }
+                var astActual = g.ToAST(converted);
+                IValue astExpected = value switch
+                {
+                    sbyte sb => new IntValue(sb),
+                    byte b => new IntValue(b),
+                    short s => new IntValue(s),
+                    ushort us => new IntValue(us),
+                    int i => new IntValue(i),
+                    uint ui => new LongValue(ui),
+                    long l => new LongValue(l),
+                    ulong ul => new BigIntValue(ul),
+                    BigInteger bi => new BigIntValue(bi),
+                    float f => new FloatValue(f),
+                    double d => new FloatValue(d),
+                    _ => null
+                };
+                astActual.ShouldBeOfType(astExpected.GetType());
+                astActual.Value.ShouldBe(astExpected.Value);
+            }
+        }
+
+        [Theory]
+        [InlineData(typeof(BooleanGraphType), true, true)]
+        [InlineData(typeof(BooleanGraphType), false, false)]
+        [InlineData(typeof(StringGraphType), "abc", "abc")]
+        [InlineData(typeof(IdGraphType), 2, 2)]
+        [InlineData(typeof(IdGraphType), "3", "3")]
+        [InlineData(typeof(FloatGraphType), 3.5, 3.5)]
+        public void parseValue_other_ok(Type graphType, object value, object parsed)
+        {
+            var g = (ScalarGraphType)graphType.GetConstructor(Type.EmptyTypes).Invoke(null);
+            var ret = g.ParseValue(value);
+            ret.ShouldBeOfType(parsed.GetType());
+            ret.ShouldBe(parsed);
+            g.CanParseValue(value).ShouldBeTrue();
```

Comment:
In general, it seems more logical in all tests first call `CanXXX` and only then `XXX`.

Reply:
I can reorder these instructions if you like.  But they all need to be tested.  We do not have a requirement that CanParse is called before Parse, and the code does not work that way.

Reply:
I know. As you see fit.